extensions/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/SelectionRenderer.java

```diff
@@ -101,42 +86,28 @@ private String getMappingsJSON(Selection w) {
-            String rowSnippet = getSnippet("selection_row");
-
-            String command = mapping.getCmd() != null ? mapping.getCmd() : "";
-            String label = mapping.getLabel();
-
-            if (item instanceof NumberItem && ((NumberItem) item).getDimension() != null) {
-                String unit = getUnitForWidget(w);
-                command = StringUtils.replace(command, UnitUtils.UNIT_PLACEHOLDER, unit);
-                label = StringUtils.replace(label, UnitUtils.UNIT_PLACEHOLDER, unit);
-
-                // Special treatment for °C since uom library uses a single character: ℃
-                // This will ensure the current state matches the cmd and the buttonClass is set accordingly.
-                command = StringUtils.replace(command, "°C", "℃");
+        if (selection.getMappings().size() == 0 && item != null && item.getStateDescription() != null) {
+            for (StateOption option : item.getStateDescription().getOptions()) {
+                jsonObject.addProperty(option.getValue(), option.getLabel());
+                rowMappingLabel = buildRow(selection, option.getLabel(), option.getValue(), item, state, rowSB);
+                if (rowMappingLabel != null) {
+                    mappingLabel = rowMappingLabel;
+                }
             }
-
-            rowSnippet = StringUtils.replace(rowSnippet, "%item%", w.getItem() != null ? w.getItem() : "");
-            rowSnippet = StringUtils.replace(rowSnippet, "%cmd%", escapeHtml(command));
-            rowSnippet = StringUtils.replace(rowSnippet, "%label%", label != null ? escapeHtml(label) : "");
-
-            State compareMappingState = state;
-            if (state instanceof QuantityType) { // convert the item state to the command value for proper
-                                                 // comparison and "checked" attribute calculation
-                compareMappingState = convertStateToLabelUnit((QuantityType<?>) state, command);
-            }
-
-            if (compareMappingState.toString().equals(command)) {
-                mappingLabel = label;
-                rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "checked=\"true\"");
-            } else {
-                rowSnippet = StringUtils.replace(rowSnippet, "%checked%", "");
+        } else {
+            for (Mapping mapping : selection.getMappings()) {
+                jsonObject.addProperty(mapping.getCmd(), mapping.getLabel());
+                rowMappingLabel = buildRow(selection, mapping.getLabel(), mapping.getCmd(), item, state, rowSB);
+                if (rowMappingLabel != null) {
+                    mappingLabel = rowMappingLabel;
+                }
             }
-            rowSB.append(rowSnippet);
         }
         snippet = StringUtils.replace(snippet, "%rows%", rowSB.toString());
+        snippet = StringUtils.replace(snippet, "%value_map%", StringEscapeUtils.escapeHtml(jsonObject.toString()));
```

Comment:
I you'd continue to use the inherited `escapeHtml`, it would be easier to refactor, if we want to replace apache commons one day.

Reply:
I changed nothing, the code was just moved from the method I suppressed (as requested in another review comment).
I don't know if there was a valid reason to use antoher function.

Reply:
Ok, it isn't identical to the line before the move, but you are right that this class was used before already. So nothing that must be fixed in the scope of this PR.